workshops/forms.py

```diff
@@ -1080,3 +1107,244 @@ class Meta:
+
+    helper.layout = Layout(
+        # no 'trainees' -- you should take care of generating it manually in
+        # the template where this form is used
+
+        # We use formnovalidate on submit button to disable browser
+        # validation. This is necessary because this form is used along with
+        # BulkAddTrainingProgressForm, which have required fields. Both forms
+        # live inside the same <form> tag. Without this attribute, when you
+        # click the following submit button, the browser reports missing
+        # values in required fields in BulkAddTrainingProgressForm.
+        FormActions(
+            Submit('discard',
+                   'Bulk discard all progresses of selected trainees',
+                   css_class='btn-danger',
+                   formnovalidate='formnovalidate'),
+        )
+    )
+    helper.inputs[:] = []  # remove default submit button
+    helper.form_tag = False
+
+    class Meta:
+        model = TrainingProgress
+
+
+class BulkChangeTrainingRequestForm(forms.Form):
+    """Form used to bulk discard training requests or bulk unmatch trainees
+    from trainings."""
+
+    requests = forms.ModelMultipleChoiceField(
+        queryset=TrainingRequest.objects.all())
+
+    helper = BootstrapHelper()
+    helper.layout = Layout(
+        # no 'requests' -- you should take care of generating it manually in
+        # the template where this form is used
+
+        # We use formnovalidate on submit buttons to disable browser
+        # validation. This is necessary because this form is used along with
+        # BulkMatchTrainingRequestForm, which have required fields. Both
+        # forms live inside the same <form> tag. Without this attribute,
+        # when you click one of the following submit buttons, the browser
+        # reports missing values in required fields in
+        # BulkMatchTrainingRequestForm.
+        FormActions(
+            Submit('discard', 'Bulk discard',
+                   formnovalidate='formnovalidate'),
+            Submit('unmatch', 'Bulk unmatch people from training',
+                   formnovalidate='formnovalidate'),
+            HTML('<a onclick="bulk_email();" class="btn btn-primary">'
+                 'Mail selected people</a>&nbsp;'),
+            HTML('<a href="{% url \'training_request\' %}" class="btn btn-success">'
+                 'New training request</a>'),
+        )
+    )
+    helper.inputs[:] = []  # remove default submit button
+
+    helper.form_tag = False
+
+    # When set to True, the form is valid only if every request is matched to
+    # one person. Set to True when 'unmatch' button is clicked, because
+    # unmatching makes sense only if each selected TrainingRequest is matched
+    # with one person.
+    check_person_matched = False
+
+    def clean(self):
+        super().clean()
+        if (self.check_person_matched and
+            any(r.person is None for r in self.cleaned_data.get('requests', []))):
```

Comment:
Assign the `any()` part to a variable and simplify writing of this conditional?

Also, is there a test that checks all of these conditions here?